internal/planmodifiers/attribute.go

```diff
@@ -117,10 +117,10 @@ func (d *numberNumericAttributePlanModifier) Modify(ctx context.Context, req tfs
 		return
 	}
 
-	if !numberConfig.Null && !numericConfig.Null {
+	if !numberConfig.Null && !numericConfig.Null && (numberConfig.Value != numericConfig.Value) {
 		resp.Diagnostics.AddError(
-			"Number numeric attribute plan modifier failed",
-			"Cannot specify both number and numeric in config",
+			"Number and numeric are both present and have different values",
```

Comment:
Tiny nit to avoid "and" twice 😄 

```suggestion
			"Number and numeric are both configured with different values",
```